### DISCUSSION TITLE
my test of  duckdb vs clickhouse vs databend

### DISCUSSION THREAD

**Discussion by l1t1** [duckdb/duckdb#7393] [General]
test on a PC with   8core CPU 8GB memory linux(free mem 6GB) | duckdb | clickhouse | databend
-- | -- | -- | --
query a range |   |   |  
SELECT   avg(i) FROM range(1,1000000000)t(i) group by round(log10(i)); | 64.068 |   | 36.979
SELECT avg(number) FROM   numbers(1,1000000000)t group by round(log(number)); |   | 20.243 |  
SELECT avg(number) FROM   numbers_mt(1,1000000000)t group by round(log(number)); |   | 2.691 |  
SELECT avg(number) FROM   numbers(1000000000)t group by round(log(number)); |   |   | 5.475
SELECT avg(number) FROM   numbers_mt(1000000000)t group by round(log(number)); |   |   | 5.634
  |   |   |  
query a mem table |   |   |  
create table t as select i::int i from   range(1,1000000000)t(i); | 9.898 |   |  
create table t engine=Memory as SELECT   number::int i FROM numbers(1,1000000000)t; |   | 6.451 |  
create table t engine=memory as SELECT   i::int i FROM numbers(1000000000)t(i); |   |   | 1.275
SELECT avg(i) FROM t group by   round(log10(i)); | 9.336 | 5.742 | 5.351
SELECT avg(i) FROM t group by   round(ln(i)) | 8.378 |   |  
SELECT avg(i) FROM t group by   round(log(i)); |   | 2.745 | 4.998
  |   |   |  
query a file table |   |   |  
create table t as select i::int i from   range(1,1000000000)t(i); | 21.786 |   |  
create table t engine=MergeTree order by   i as SELECT number::int i FROM numbers(1,1000000000)t; |   | 85.022 |  
create table t storage_format='native'   compression='lz4' as SELECT i::int i FROM numbers(1000000000)t(i); |   |   | 126.15
SELECT avg(i) FROM t group by   round(log10(i)); | 9.246 | 7.602 | 6.525
SELECT avg(i) FROM t group by   round(log(i)); |   | 3.615 | 6.339
  |   |   |  
query a temp directory table |   |   |  
set temp_directory='tmp.tmp';set   memory_limit='5GB'; | 8.995 |   |  
create table t engine=Memory as SELECT   number i FROM numbers(1,1000000000)t; |   | 23.458 |  
SELECT avg(i) FROM t group by   round(log10(i)); | 25.524 | 36.979 |

**Comment by marsupialtail** [comment]
Why are there empty cells in the table?

**Comment by l1t1** [comment]
because some of the same meaning sql of different db are different. 
eg. duckdb has no range function

**Comment by l1t1** [comment]
add more tests
https://github.com/datafuselabs/databend/discussions/11357
